test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,301 @@
+
+/**
+ * @given torii service
+ *        and some number of valid transactions
+ *        and one stateless invalid tx
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus publishes statelessInvalid for all txes
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiPartialInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request{};
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  int counter = 0;
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Invoke([&counter](const auto &) mutable {
+        shared_model::validation::Answer res;
+        if (counter++ == kTimes - 1) {
+          res.addReason(
+              std::make_pair("some error", std::vector<std::string>{}));
+        }
+        return res;
+      }));
+  EXPECT_CALL(
+      *batch_factory,
+      createTransactionBatch(
+          A<const shared_model::interface::types::SharedTxsCollectionType &>()))
+      .Times(kTimes - 1);
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes - 1);
+  EXPECT_CALL(*status_bus, publish(_)).WillOnce(Invoke([](auto status) {
+    EXPECT_FALSE(status->statelessErrorOrCommandName().empty());
+  }));
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and command_service with empty status stream
+ * @when calling StatusStream on transport
+ * @then Ok status is eventually returned without any fault
+ */
+TEST_F(ToriiTransportCommandTest, StatusStreamEmpty) {
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+
+  EXPECT_CALL(*command_service, getStatusStream(_))
+      .WillOnce(Return(rxcpp::observable<>::empty<std::shared_ptr<
+                           shared_model::interface::TransactionResponse>>()));
+
+  ASSERT_TRUE(transport_grpc->StatusStream(&context, &request, nullptr).ok());
+}
+
+/**
+ * @given torii service and number of invalid transactions
+ * @when calling StatusStream
+ * @then ServerWriter call Write method the same number of times
+ */
+TEST_F(ToriiTransportCommandTest, StatusStream) {
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+  iroha::MockServerWriter<iroha::protocol::ToriiResponse> response_writer;
+
+  std::vector<std::shared_ptr<shared_model::interface::TransactionResponse>>
+      responses;
+  for (size_t i = 0; i < kTimes; ++i) {
+    shared_model::crypto::Hash hash(std::to_string(i));
+    auto push_response = [this, hash = std::move(hash), &responses](
+                             auto member_fn) {
+      responses.emplace_back((this->status_factory.get()->*member_fn)(
+                                 hash, {} /*"ErrMessage" + std::to_string(i)*/)
+                                 .release());
+    };
+
+    using shared_model::interface::TxStatusFactory;
+    // cover different type of statuses
```

Comment:
What is the point of calling different methods if there is no assertion that correct corresponding messages are being written to status stream?

Reply:
To ensure that all of the types handled in the same way

Reply:
But this logic is specified in test. How does it test `CommandServiceTransportGrpc` class?

Reply:
I'm not sure, that I got your question. It test via calling `StatusStream` below. Different status types may lead to the different execution paths, that's the point here.

Reply:
It tests a single execution path - getTransport, so there is no need to check more than one status type.